sql/src/main/java/io/crate/analyze/GenericPropertiesConverter.java

```diff
@@ -70,26 +72,65 @@ static Settings genericPropertiesToSettings(GenericProperties genericProperties,
         return builder.build();
     }
 
+
+    static Settings.Builder settingsFromProperties(GenericProperties properties,
+                                                   Row parameters,
+                                                   Map<String, Setting> supportedSettings) {
+
+        return settingsFromProperties(properties, parameters, supportedSettings, true);
+    }
+
     public static Settings.Builder settingsFromProperties(GenericProperties properties,
-                                                          ParameterContext parameterContext,
-                                                          Map<String, ? extends SettingsApplier> settingAppliers) {
+                                                          Row parameters,
+                                                          Map<String, Setting> supportedSettings,
+                                                          boolean setDefaults) {
+
         Settings.Builder builder = Settings.builder();
-        setDefaults(settingAppliers, builder);
+        settingsFromProperties(builder, properties, parameters, supportedSettings, setDefaults, (s) -> false, INVALID_SETTING_MESSAGE);
+        return builder;
+    }
+
+    static void settingsFromProperties(Settings.Builder builder,
+                                       GenericProperties properties,
+                                       Row parameters,
+                                       Map<String, Setting> supportedSettings,
+                                       boolean setDefaults,
+                                       Predicate<String> ignoreProperty,
+                                       String invalidMessage) {
```

Comment:
is having the invalidMessage as parameter of any value? (maybe I'm missing some usages but is this ever different?)